core/src/main/java/com/graphhopper/routing/util/AreaIndex.java

```diff
@@ -42,48 +45,54 @@ default Polygon getBorder() {
     private final GeometryFactory gf;
     private final STRtree index;
 
+    private final PreparedGeometryFactory pgf;
+
+    private final DistanceCalc dc;
+
     public AreaIndex(List<T> areas) {
         gf = new GeometryFactory();
+        pgf = new PreparedGeometryFactory();
+        dc = new DistanceCalcEarth();
         index = new STRtree();
-        PreparedGeometryFactory pgf = new PreparedGeometryFactory();
         for (T area : areas) {
             for (Polygon border : area.getBorders())
-                addBorder(pgf, area, border);
+                addBorder( area, border);
             if (area.getBorder() != null)
-                addBorder(pgf, area, area.getBorder());
+                addBorder( area, area.getBorder());
         }
         index.build();
     }
 
-    private void addBorder(PreparedGeometryFactory pgf, T area, Polygon border) {
-        // todonow: do we really need the prepared geometry for osm areas?
-        IndexedCustomArea<T> indexedCustomArea = new IndexedCustomArea<>(area, pgf.create(border));
-        index.insert(border.getEnvelopeInternal(), indexedCustomArea);
+    private void addBorder( T area, Polygon border) {
+        IndexedArea<T> indexedOsmArea = new IndexedArea<>(area, border);
+        index.insert(border.getEnvelopeInternal(), indexedOsmArea);
     }
 
     public List<T> query(double lat, double lon) {
-        Envelope searchEnv = new Envelope(lon, lon, lat, lat);
+        BBox bbox = dc.createBBox(lon, lat, 10);
```

Comment:
This should be `lat, lon, 10` not `lon, lat, 10`, no?

Reply:
Thanks for detecting this bug! I ran a new benchmark with the corrected order of the parameters and got following result: `[main] INFO  org.eclipse.jetty.server.Server - Started @596397ms`, which is just a bit slower.